llvm/lib/Target/AMDGPU/SIFoldOperands.cpp

```diff
@@ -2246,3 +2255,14 @@ bool SIFoldOperands::runOnMachineFunction(MachineFunction &MF) {
 
   return Changed;
 }
+
+PreservedAnalyses SIFoldOperandsPass::run(MachineFunction &MF, MachineFunctionAnalysisManager&) {
+  SIFoldOperandsImpl Impl;
+  bool Changed = Impl.run(MF);
+  if(!Changed) {
+    return PreservedAnalyses::all();
+  }
+  PreservedAnalyses PA;
```

Comment:
Better use `auto PA = getMachineFunctionPassPreservedAnalyses();`

Reply:
Updated in the new PR